nuxeo-runtime/nuxeo-runtime-stream/src/test/java/org/nuxeo/runtime/stream/RuntimeStreamFeature.java

```diff
@@ -0,0 +1,120 @@
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.RunnerFeature;
+import org.nuxeo.runtime.test.runner.RuntimeFeature;
+import org.nuxeo.runtime.test.runner.RuntimeHarness;
+import org.osgi.framework.Bundle;
+
+/**
+ * The runtime stream feature provides a Chronicle or Kafka stream implementation depending on test configuration.
+ * <p/>
+ * To run your unit tests on a Chronicle or Kafka you need to declare {@code nuxeo.test.stream.type} to either
+ * {@link #STREAM_CHRONICLE chronicle} or {@link #STREAM_KAFKA kafka} in your system properties.
+ * <p/>
+ * Once your tests use this feature, you can deploy new {@code logConfig} like below:
+ * <pre>
+ * {@code <extension target="org.nuxeo.runtime.stream.service" point="logConfig">
+ *     <logConfig name="MY_LOG_CONFIG_NAME" type="${nuxeo.test.stream.type}" />
+ *   </extension>
+ * }
+ * </pre>
+ *
+ * @since 10.3
+ */
+@Deploy("org.nuxeo.runtime.stream")
+@Deploy("org.nuxeo.runtime.stream.test")
+@Features(RuntimeFeature.class)
+public class RuntimeStreamFeature implements RunnerFeature {
+
+    private static final Log log = LogFactory.getLog(RuntimeStreamFeature.class);
+
+    public static final String BUNDLE_TEST_NAME = "org.nuxeo.runtime.stream.test";
+
+    public static final String STREAM_PROPERTY = "nuxeo.test.stream.type";
+
+    public static final String STREAM_CHRONICLE = "chronicle";
+
+    public static final String STREAM_KAFKA = "kafka";
+
+    // kafka properties part
+
+    public static final String KAFKA_SERVERS_PROPERTY = "nuxeo.test.kafka.servers";
+
+    public static final String KAFKA_SERVERS_DEFAULT = "localhost:9092";
+
+    protected String streamType;
+
+    // TODO copied from StorageConfiguration
+    protected static String defaultProperty(String name, String def) {
+        String value = System.getProperty(name);
+        if (value == null || value.equals("") || value.equals("${" + name + "}")) {
+            value = def;
+        }
+        Framework.getProperties().setProperty(name, value);
+        return value;
+    }
+
+    @Override
+    public void start(FeaturesRunner runner) {
+        RuntimeHarness harness = runner.getFeature(RuntimeFeature.class).getHarness();
+        streamType = defaultProperty(STREAM_PROPERTY, STREAM_CHRONICLE);
+        try {
+            switch (streamType) {
+            case STREAM_CHRONICLE:
+                initChronicle(harness);
+                break;
+            case STREAM_KAFKA:
+                initKafka(harness);
+                break;
+            default:
+                throw new UnsupportedOperationException(streamType + " stream type is not supported");
+            }
+        } catch (IOException e) {
+            throw new RuntimeServiceException("Unable to configure the stream implementation", e);
+        }
+    }
+
+    protected void initChronicle(RuntimeHarness harness) {
+        log.info("Deploying Chronicle for stream services");
+        // nothing to do
```

Comment:
is that really necessary ?

Reply:
For the log and equivalence with kafka I would say yes.

Reply:
log.info is most of the time disabled, you should do like with repo backend or elasticsearch client selection: write to stdout the selected config for the test when it is not the default wm.

Reply:
I was thinking about a common logger configured for such thing and for _Platform Started_ message.

Reply:
Let's discuss about this idea, in the meantime I will change them to stdout.